pkg/imgpkg/bundle/images_lock.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package bundle
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/cppforlife/go-cli-ui/ui"
+	regname "github.com/google/go-containerregistry/pkg/name"
+	ctlimg "github.com/k14s/imgpkg/pkg/imgpkg/image"
+	"github.com/k14s/imgpkg/pkg/imgpkg/lockconfig"
+)
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 github.com/cppforlife/go-cli-ui/ui.UI
+
+func NewImagesLock(imagesLock *lockconfig.ImagesLock, imgRetriever ctlimg.ImagesMetadata, relativeToRepo string) *ImagesLock {
+	return &ImagesLock{imagesLock: imagesLock, imgRetriever: imgRetriever, relativeToRepo: relativeToRepo}
+}
+
+type ImagesLock struct {
+	imagesLock     *lockconfig.ImagesLock
+	imgRetriever   ctlimg.ImagesMetadata
+	relativeToRepo string
+}
+
+func (o *ImagesLock) WriteToPath(outputPath string, ui ui.UI) error {
+	imagesLockPath := filepath.Join(outputPath, ImgpkgDir, ImagesLockFile)
+	ui.BeginLinef("Locating image lock file images...\n")
+
+	skipped, err := o.localizeImagesLock()
+	if err != nil {
+		return err
+	}
+	if skipped {
+		ui.BeginLinef("One or more images not found in bundle repo; skipping lock file update\n")
+	} else {
+		ui.BeginLinef("The bundle repo (%s) is hosting every image specified in the bundle's Images Lock file (.imgpkg/images.yml)\n", o.relativeToRepo)
+		ui.BeginLinef("\nUpdating all images in the ImagesLock file: %s\n", imagesLockPath)
+		ui.BeginLinef("+ Changing all image registry/repository references in %s to %s\n", imagesLockPath, o.relativeToRepo)
+	}
+	return o.imagesLock.WriteToPath(imagesLockPath)
+
+}
+
+func (o *ImagesLock) localizeImagesLock() (bool, error) {
```

Comment:
i see that we are using this private method in Bundle object. i would recommend turning this into `Localized() (*lockconfig.ImagesLock, bool, error)` i.e just return new lockconfig.ImagesLock with localized images.